### ISSUE TITLE
potentiall failure in eval notebook

### ISSUE THREAD

**Issue by stevehadd** [informatics-lab/ukrse_2022_mlops_walkthrough#25]
comment from harvey in review

or this cell (where confusion matrix is calculated) and above, in cases where a conf mat was generated for a batch that only had one label, the dimensions of the output from the sklearn.metrics.confusion_matrix function became [1,1] instead of [2,2]. Passing the label argument to the function ensures shape. I assume this won't be an encountered issue, but had an exception while running through while the data for rotors had NaN values (commented in 02)

numpy.unique(y_val).astype(int) is the first implemented way of generating the labels, so could do with some further consideration in case of edge case failure
